packages/postgres/src/pool.ts

```diff
@@ -924,7 +927,23 @@ export class PostgresPool {
           client.release();
         }
       },
+      stream: function (batchSize: number) {
+        const transform = new Transform({
+          readableObjectMode: true,
+          writableObjectMode: true,
+          transform(chunk, _encoding, callback) {
+            for (const row of chunk) {
+              this.push(row);
+            }
+            callback();
+          },
+        });
+
+        // TODO: use native `node:stream#compose` once it's stable.
+        return multipipe(Readable.from(iterator.iterate(batchSize)), transform);
```

Comment:
`pipeline` is best suited for when you have an initial source and a final destination; `multipipe` (or `compose`, which it mirrors) lets you construct a stream that in turn will be piped elsewhere. I think the [`compose` docs](https://nodejs.org/api/stream.html#streamcomposestreams) say this best:

> Because stream.compose returns a new stream that in turn can (and should) be piped into other streams, it enables composition. In contrast, when passing streams to stream.pipeline, typically the first stream is a readable stream and the last a writable stream, forming a closed circuit.

Reply:
Cool, thanks!